test/functional/test_runner.py

```diff
@@ -219,9 +224,12 @@ def main():
 
     # Remove the test cases that the user has explicitly asked to exclude.
     if args.exclude:
-        for exclude_test in args.exclude.split(','):
-            if exclude_test + ".py" in test_list:
-                test_list.remove(exclude_test + ".py")
+        tests_excl = [re.sub("\.py$", "", t) + ".py" for t in args.exclude.split(',')]
+        for exclude_test in tests_excl:
+            if exclude_test in test_list:
+                test_list.remove(exclude_test)
```

Comment:
Tthis will only remove the first instance of `exclude_test` from `test_list`. It's now possible that there are multiple instances of the script from line 176 above. Something like this will work:

    test_list = filter(lambda t: t == exclude_test, test_list)